scripts/gen_enums.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+
+warning = """
+/*
+ * WARNING: This file is autogenerated from scripts/gen_enums.py. If you would
+ * like to access an enum that is currently missing, add it to the script
+ * and run it from the root directory to update this file.
+ */
+
+"""
+
+enums = [
+            ("scx_public_consts", "SCX_OPS_NAME_LEN"),
+            ("scx_public_consts", "SCX_SLICE_DFL"),
+            ("scx_public_consts", "SCX_SLICE_INF"),
+
+            ("scx_dsq_id_flags", "SCX_DSQ_FLAG_BUILTIN"),
+            ("scx_dsq_id_flags", "SCX_DSQ_FLAG_LOCAL_ON"),
+            ("scx_dsq_id_flags", "SCX_DSQ_INVALID"),
+            ("scx_dsq_id_flags", "SCX_DSQ_GLOBAL"),
+            ("scx_dsq_id_flags", "SCX_DSQ_LOCAL"),
+            ("scx_dsq_id_flags", "SCX_DSQ_LOCAL_ON"),
+            ("scx_dsq_id_flags", "SCX_DSQ_LOCAL_CPU_MASK"),
+
+            ("scx_ent_flags", "SCX_TASK_QUEUED"),
+            ("scx_ent_flags", "SCX_TASK_RESET_RUNNABLE_AT"),
+            ("scx_ent_flags", "SCX_TASK_DEQD_FOR_SLEEP"),
+            ("scx_ent_flags", "SCX_TASK_STATE_SHIFT"),
+            ("scx_ent_flags", "SCX_TASK_STATE_BITS"),
+            ("scx_ent_flags", "SCX_TASK_STATE_MASK"),
+            ("scx_ent_flags", "SCX_TASK_CURSOR"),
+
+            ("scx_task_state", "SCX_TASK_NONE"),
+            ("scx_task_state", "SCX_TASK_INIT"),
+            ("scx_task_state", "SCX_TASK_READY"),
+            ("scx_task_state", "SCX_TASK_ENABLED"),
+            ("scx_task_state", "SCX_TASK_NR_STATES"),
+
+            ("scx_ent_dsq_flags", "SCX_TASK_DSQ_ON_PRIQ"),
+
+            ("scx_kick_flags", "SCX_KICK_IDLE"),
+            ("scx_kick_flags", "SCX_KICK_PREEMPT"),
+            ("scx_kick_flags", "SCX_KICK_WAIT"),
+
+            ("scx_enq_flags", "SCX_ENQ_WAKEUP"),
+            ("scx_enq_flags", "SCX_ENQ_HEAD"),
+            ("scx_enq_flags", "SCX_ENQ_PREEMPT"),
+            ("scx_enq_flags", "SCX_ENQ_REENQ"),
+            ("scx_enq_flags", "SCX_ENQ_LAST"),
+            ("scx_enq_flags", "SCX_ENQ_CLEAR_OPSS"),
+            ("scx_enq_flags", "SCX_ENQ_DSQ_PRIQ"),
```

Comment:
That would make maintaining the autogen'ed files significantly easier, but I think it would be better to do it in a separate diff on top because this one is getting on the large side